ambry-account/src/main/java/com/github/ambry/account/InMemoryAccountService.java

```diff
@@ -0,0 +1,146 @@
+    this.accountFile = accountFile;
+  }
+
+  /** Guards against calling {@link #init} multiple times.*/
+  private final AtomicBoolean initialized = new AtomicBoolean();
+
+  /**
+   * Initializes this instance. Does all work which requires handing out references to this instance itself which can't
+   * be done during normal instance construction without letting a reference to an incomplete constructed instance escape.
+   */
+  public void init() {
+    if (!initialized.getAndSet(true)) {
+      processAccountJsonFile();
+
+      // add the default unknown account
+      ArrayList<Account> accounts = new ArrayList<Account>();
+      accounts.add(UNKNOWN_ACCOUNT);
+
+      try {
+        updateAccounts(accounts);
+      } catch (AccountServiceException e) {
+        logger.warn("Unable to add default unknown account", e);
+      }
+
+      }
+  }
+
+  @Override
+  protected void checkOpen() {
+    if (!initialized.get()) {
+      throw new IllegalStateException("AccountService is closed.");
+    }
+  }
+
+  @Override
+  protected void onAccountChangeMessage(String topic, String message) {
+    // We're good.
+  }
+
+  /**
+   * Adds accounts
+   *
+   * @param accounts The collection of {@link Account}s to update. Cannot be {@code null}.
+   */
+  @Override
+  public void updateAccounts(Collection<Account> accounts) throws AccountServiceException {
+    if (accountInfoMapRef.get().hasConflictingAccount(accounts, true)) {
+      throw new AccountServiceException("Input accounts conflict with the accounts in local cache",
+          AccountServiceErrorCode.ResourceConflict);
+    }
+    for (Account account : accounts) {
+      if (accountInfoMapRef.get().hasConflictingContainer(account.getAllContainers(), account.getId(), true)) {
+        throw new AccountServiceException("Containers in account " + account.getId() + " conflict with local cache",
+            AccountServiceErrorCode.ResourceConflict);
+      }
+    }
+    accountInfoMapRef.get().addOrUpdateAccounts(accounts);
+  }
+
+  @Override
+  public void close() throws IOException {
+  }
+
+  /**
+   * Loads initial accounts into memory via a JSON file.
+   */
+  private void processAccountJsonFile() {
+    if (accountFile == null) {
+      return;
+    }
+
+    try {
+      if (!Files.exists(accountFile)) {
+        logger.warn("JSON account file '{}' does not exist", accountFile.toAbsolutePath());
+        return;
+      }
+      String accountsJsonString = new String(Files.readAllBytes(accountFile), StandardCharsets.UTF_8);
+      AccountInfoMap newAccountInfoMap = new AccountInfoMap(accountsJsonString);
+      AccountInfoMap oldAccountInfoMap = accountInfoMapRef.getAndSet(newAccountInfoMap);
+
+      //Notify modified accounts to consumers
+      Collection<Account> updatedAccounts = newAccountInfoMap.getAccounts()
+          .stream()
+          .filter(newAccount -> !newAccount.equals(oldAccountInfoMap.getAccountById(newAccount.getId())))
```

Comment:
processAccountJsonFile should initialize the account map in `AccountInfoMap`, which means there is not `oldAccountInfoMap` here, you don't have to do this `filtering`.